extensions/BugModal/lib/WebService.pm

```diff
@@ -62,6 +81,25 @@ sub rest_resources {
     ]
 }
 
+sub products {
+    my $user = Bugzilla->user;
+    my @products = @{ $user->get_enterable_products };
+
+    @products = map { { name => $_->name } } @products;
+    return {
+        products     => \@products,
+    };
+}
+
+sub components {
+    my ($self, $params) = @_;
+    my $product = Bugzilla::Product->check({ name => $params->{product_name}, cache => 1 });
+    my $components = _name($product->components);
+    my %options;
+    $options{component} = $components;
+    return \%options;
```

Comment:
Well, the components one doesn't obviously document the resulting output object. It requires you to implicitly understand what _name() returns, which I don't know, and so I couldn't predict what this will return from the API without reading _name().

You do use two different methods to construct the same output, which misleads me into thinking that you're constructing two different types of output. I do think it would be useful to make the two code paths *look* more similar, since the code is the only documentation available for the resulting data structure.